src/libraries/System.Net.Sockets/tests/FunctionalTests/TelemetryTest.cs

```diff
@@ -35,16 +34,351 @@ public static void EventSource_ExistsWithCorrectId()
             Assert.NotEmpty(EventSource.GenerateManifest(esType, esType.Assembly.Location));
         }
 
+        public static IEnumerable<object[]> SocketMethods_MemberData()
+        {
+            yield return new[] { "Sync" };
+            yield return new[] { "Task" };
+            yield return new[] { "Apm" };
+            yield return new[] { "Eap" };
+        }
+
+        public static IEnumerable<object[]> SocketMethods_Matrix_MemberData()
+        {
+            return from m1 in SocketMethods_MemberData()
+                   from m2 in SocketMethods_MemberData()
+                   select new[] { m1[0], m2[0] };
+        }
+
+        public static IEnumerable<object[]> SocketMethods_WithBools_MemberData()
+        {
+            return from m1 in SocketMethods_MemberData()
+                   from @bool in new[] { true, false }
+                   select new[] { m1[0], @bool };
+        }
+
+        private static Task AcceptConnectionAsync(Socket server, string acceptMethod)
```

Comment:
Why not passing `SocketHelper` here instead of `acceptMethod`?

Reply:
Changed.